nvflare/dashboard/application/blob.py

```diff
@@ -180,6 +180,11 @@ def gen_client(key, id):
         "type": "client",
         "cln_uid": f"uid={entity.name}",
     }
+    for k in ["client_name", "org_name", "cln_uid"]:
+        value = replacement_dict[k]
+        escaped_value = value.replace("'", "\\'")
```

Comment:
That character is used for names like Joe's A100.  Because the client training starts with options including org, it causes scripts failing.
A better way is to warn users not to use any special characters, but that requires web UI changes.
I also checked the dashboard logic.  We could silently convert any special characters to _ or -.  But users could get confused because Joe's A100 would become Joe_s A100.
The provisioning CLI enforces the rules for names, so no special characters are allowed.

Reply:
"A better way is to warn users not to use any special characters, but that requires web UI changes" => should we do this better way for main branch?

If we want we could have this patch in 2.4?

Reply:
I think we don't have to patch it on 2.4 branch because the only issue is when users enter org name with ``'``  The web UI change requires adding some regex check in existing js codes.  Maybe @nvkevlu can help.